mm/mm_heap/mm_malloc.c

```diff
@@ -244,15 +244,18 @@ FAR void *mm_malloc(FAR struct mm_heap_s *heap, size_t size)
 #ifdef CONFIG_DEBUG_MM
   else
     {
+#ifdef CONFIG_MM_DUMP_ON_FAILURE
       struct mallinfo minfo;
+#endif
 
       mwarn("WARNING: Allocation failed, size %zu\n", alignsize);
+#ifdef CONFIG_MM_DUMP_ON_FAILURE
       mm_mallinfo(heap, &minfo);
       mwarn("Total:%d, used:%d, free:%d, largest:%d, nused:%d, nfree:%d\n",
             minfo.arena, minfo.uordblks, minfo.fordblks,
             minfo.mxordblk, minfo.aordblks, minfo.ordblks);
       mm_memdump(heap, -1);
-      DEBUGASSERT(false);
+#endif
```

Comment:
```suggestion
      DEBUGPANIC();
#endif
```
Was the removal of the `DEBUGASSERT(false);` intended?
I believe leaving it here may be valuable during debugging.

Reply:
Ok, just noticed the separate commit just for this change. It makes sense.

Reply:
where are the separate commit?

Reply:
but DEBUGASSERT get removed?

Reply:
Yes. There is an explanation in commit message. Maybe debug assert should be optional, I do not know.

Reply:
DEBUGPANIC/DEBUGASSERT could generate the callstack which is very useful to know the abnormal location, so it's better to keep the as before. If user want to see heap info, he should also want to see the callback info too.

Reply:
ok. i added it back as a separate option.